subprojects/testfx-core/src/main/java/org/testfx/util/NodeQueryUtils.java

```diff
@@ -250,7 +252,7 @@ private static boolean matchesNodeMatcher(Node node,
 
     @SuppressWarnings("deprecation")
     private static boolean isNodeVisible(Node node) {
-        if (!node.isVisible() || !node.impl_isTreeVisible()) {
+        if (isNotVisible(node)) {
```

Comment:
I'd guess it only matters if one is using a Tree-related control, a component which I'm not sure we test for in TestFX or not. For all we know, they could have implemented whatever code that was into the Tree node itself... Otherwise, yes, outside of our control